### ISSUE TITLE
Create an article about the upgrade process from one version to another

### ISSUE THREAD

**Issue by valentinstoychev** [NativeScript/NativeScript#316]
We need to document what is the process a developer should do when upgrading from one version to another.

**Comment by meal** [comment]
where can we find such information at this moment? is it matter of reviewing commits and finding out what has been changed?

**Comment by valentinstoychev** [comment]
@meal for 1.1. we are still preparing the official announcement with all the details you need. Meanwhile you can see the [CHANGELOG](https://github.com/NativeScript/NativeScript/blob/master/CHANGELOG.md). 

Please stay tuned to see all the details in a blog post and also we will host a webinar dedicated on the 1.1 release.

**Comment by ligaz** [comment]
Here is way you can upgrade your existing projects to 1.1:

### OS X / Linux
```bash
$ tns platform update ios
$ npm install https://github.com/NativeScript/NativeScript/releases/download/v1.1.0/tns-core-1.1.0.tgz
$ rsync -a node_modules/tns-core/ app/tns_modules
$ rm -rf node_modules/tns-core/
```

### Windows
```bash
$ tns platform update android
$ npm install https://github.com/NativeScript/NativeScript/releases/download/v1.1.0/tns-core-1.1.0.tgz
$ xcopy .\node_modules\tns-core\* .\app\tns_modules /s /Y /i
$ del /S /Q node_modules\tns-core\*
```

**Comment by mdgiles** [comment]
Thanks for this. What would be the way to make sure typescript definitions are at the same level as well?

**Comment by Nathanaela** [comment]
Two things;
1. you still should upgrade your nativescript tool:
**npm install -g nativescript**

2. these instructions will not work on windows xp, 2003 or Vista unless you download robocopy from the microsoft resource kits.  I have outlined alternative upgrade steps last week which should work for all os's - http://fluentreports.com/blog/?p=79.

**Comment by ligaz** [comment]
Hey @Nathanaela,

The NativeScript CLI is officially support for Vista and above. The last time I check [robocopy comes with Vista](https://en.wikipedia.org/wiki/Robocopy) :smile: 

Thanks for providing your steps as well. However I don't think they accommodate for the case when I have custom modules added in my `tns_modules` folder.

**Comment by Nathanaela** [comment]
Interesting; I don't have a Vista virtual machine on my box so I cannot test it -- I just went with what Microsoft's site https://technet.microsoft.com/en-us/library/cc733145.aspx -- And it shows it is Win7 or newer...   But doing some googling real quick to follow up does seem like Vista has it.  Guess Microsoft's site is wrong.  LOL  :grinning:    

I personally think that would be a really bad idea adding any code to the tns_modules that you plan on keeping as that is generated from TypeScript.   So you are mixing release code with your own code your just asking for trouble. :grinning:        I might update the NS/common repo then deploy to my tns_modules; but any changes I put directly into tns_modules is by design going to be wiped out on my next update (which I do fairly frequently from master/head)    All your code should be in the node_modules folder or your app folder ---  Oh and btw, /mir is going to nuke ALL your changes too.

**Comment by FlipOneDev** [comment]
I'm sorry, but I'm a bit confused about conflicting inputs from @ligaz and @Nathanaela regarding which set of update instructions to follow. NativeScript Team, would you PLEASE CLARIFY AND FINALIZE!!

**Comment by ligaz** [comment]
Hey @FlipOneDev,

Sorry if we have mislead you with our musings. Let me summarise it for you. If you are not feeling comfortable from the command line or you want a guided upgrade guide please read the [excellent post](http://fluentreports.com/blog/?p=79) that @Nathanaela has crafted. If you want the TL;DR version please follow the [steps](https://github.com/NativeScript/NativeScript/issues/316#issuecomment-112309530) I have outlined. 

Hope this helps.

**Comment by bfattori** [comment]
I'm trying to get 1.2.0 up and running without much luck. I keep getting an error on Windows when I try to run it on the Android emulator. This is the relevant part in the command interpreter:

![image](https://cloud.githubusercontent.com/assets/697256/8967325/25f487c4-3604-11e5-9ed5-ebaaca8f3c3c.png)

It appears to be failing in trying to copy some files. I had to extract and copy in the v1.2.0 core modules according to @ligaz 's note from above (changing 1.1.0 to 1.2.0). But I also notice that the project doesn't contain the AppResources folder. Note that I started this from a clean project setup using:

`tns create my-project` and after running `tns platform add android`
